### PR TITLE
Link checker plugin and some fixes to URLs

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <mehranb@amazon.com>

### Description
Checks internal and potentially external links at build.
There are a bunch of URLs which it caught and I fixed.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by aetter — COMMENTED** [review]
Looks good, just some broken links where I think the fix is also broken.

File: _clients/logstash/ship-to-opensearch.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ nav_order: 220
 
 You can Ship Logstash events to an OpenSearch cluster and then visualize your events with OpenSearch Dashboards.
 
-Make sure you have [Logstash]({{site.url}}{{site.baseurl}}/logstash/index/#install-logstash-on-mac--linux), [OpenSearch]({{site.url}}{{site.baseurl}}/opensearch/install/index/), and [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/dashboards/install/index/).
+Make sure you have [Logstash]({{site.url}}{{site.baseurl}}/logstash/#install-logstash), [OpenSearch]({{site.url}}{{site.baseurl}}/opensearch/install/index/), and [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/dashboards/install/index/).
```
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply _clients/logstash/ship-to-opensearch.md:12]: I added
```
redirect_from:
- /logstash/
```
Would you rather I change that to this?
```
redirect_from:
- /logstash/index/
```

File: _opensearch/data-streams.md [hunk 2]
```diff
@@ -289,4 +289,4 @@ You can use wildcards to delete more than one data stream.
 
 We recommend deleting data from a data stream using an ISM policy.
 
-You can also use [asynchronous search]({{site.url}}{{site.baseurl}}/async/index/) and [SQL]({{site.url}}{{site.baseurl}}/sql/index/) and [PPL]({{site.url}}{{site.baseurl}}/ppl/index/) to query your data stream directly. You can also use the security plugin to define granular permissions on the data stream name.
+You can also use [asynchronous search]({{site.url}}{{site.baseurl}}/async/) and [SQL]({{site.url}}{{site.baseurl}}/sql/) and [PPL]({{site.url}}{{site.baseurl}}/ppl/) to query your data stream directly. You can also use the security plugin to define granular permissions on the data stream name.
```
[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply _opensearch/data-streams.md:292]: These are also available due to `redirect_from` values of `/sql/`, `/async/`, and `/ppl/`. Would you rather I remove those redirects and use the links you provided?

File: _clients/logstash/ship-to-opensearch.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aetter) [reply _clients/logstash/ship-to-opensearch.md:12]: Yeah, overall, I think we want the links we create to be reflective of the actual path and use redirects for page moves/index pages/links that people might typo.

[the remark this answers is not in the document]
Comment (AMoo-Miki) [reply _clients/logstash/ship-to-opensearch.md:12]: Of course. Will take care of all 4

**Review by aetter — APPROVED** [review]

**Pull request merged**